### PR TITLE
Add runAsUser/runAsGroup fields to ActiveGate SecurityContext

### PR DESCRIPTION
**Pull request by aorcholski**
# Description

ActiveGate containers i.e.  `activegate` regular container and `certificate-loader` init container should have the same securityContext. Additionally the securityContext should have `runAsUser` and `runAsGroup` fields set to the same value as it is specified by ActiveGate Dockerfile (`USER`).

## How can this be tested?
Deploy DynaKube with `kubernetes-monitoring` capability enabled on:
- a Kubernetes cluster
- an OpenShift cluster.

## Checklist
- [x] Unit tests have been updated/added
- [x] PR is labeled accordingly
- [x] I have read and understood the [contribution guidelines](https://github.com/Dynatrace/dynatrace-operator/blob/master/CONTRIBUTING.md)

### CODE REVIEW

File: src/controllers/dynakube/activegate/consts/consts.go [hunk 1]
```diff
@@ -59,4 +61,26 @@ const (
 	GatewaySslMountPoint     = "/var/lib/dynatrace/gateway/ssl"
 	GatewayLogMountPoint     = "/var/log/dynatrace/gateway"
 	GatewayTmpMountPoint     = "/var/tmp/dynatrace/gateway"
+
+	DockerImageUser  int64 = 1001
+	DockerImageGroup int64 = 1001
+)
+
+var (
+	ContainerSecurityContext = corev1.SecurityContext{
```
Comment (luhi-DT) [inline src/controllers/dynakube/activegate/consts/consts.go:70]: I would put that into a method of the kubemon modifier like `getSecurityContext()`

File: src/controllers/dynakube/activegate/internal/statefulset/builder/modifiers/kubemon.go [hunk 2]
```diff
@@ -72,11 +73,11 @@ func (mod KubernetesMonitoringModifier) getInitContainers() []corev1.Container {
 			Args:            []string{"-c", k8scrt2jksPath},
 			VolumeMounts:    volumeMounts,
 			Resources:       mod.capability.Properties().Resources,
-			SecurityContext: &corev1.SecurityContext{
-				ReadOnlyRootFilesystem: &readOnlyRootFs,
-			},
+			SecurityContext: consts.ContainerSecurityContext.DeepCopy(),
```
Comment (luhi-DT) [inline src/controllers/dynakube/activegate/internal/statefulset/builder/modifiers/kubemon.go:76]: Did you test if it is fine using the same security context on the container as on the pod?

**Review by luhi-DT — CHANGES_REQUESTED** [review]

File: src/controllers/dynakube/activegate/internal/statefulset/builder/modifiers/kubemon.go (hunk 2 above)
Comment (aorcholski) [reply src/controllers/dynakube/activegate/internal/statefulset/builder/modifiers/kubemon.go:76]: Only some of ContainerSecurityContext settings can be also set on the pod level
```
	PodSecurityContext = corev1.PodSecurityContext{
		// Privileged:               address.Of(false),
		// AllowPrivilegeEscalation: address.Of(false),
		RunAsNonRoot: address.Of(true),
		RunAsUser:    address.Of(DockerImageUser),
		RunAsGroup:   address.Of(DockerImageGroup),
		// Capabilities: &corev1.Capabilities{
		//	Drop: []corev1.Capability{
		//		"ALL",
		//	},
		// },
		SeccompProfile: &corev1.SeccompProfile{
			Type: corev1.SeccompProfileTypeRuntimeDefault,
		},
		//ReadOnlyRootFilesystem: address.Of(false),
	}
```
so I decided to keep them all in the shared ContainerSecurityContext structure.

I have tested a case with `PodSecurityContext` defined and istio enabled. Everything is fine (istio containers define own SecurityContexts with `runAsUser: 0` or `runAsUser: 1337`).

File: src/controllers/dynakube/activegate/consts/consts.go [hunk 3]
```diff
@@ -59,4 +59,7 @@ const (
 	GatewaySslMountPoint     = "/var/lib/dynatrace/gateway/ssl"
 	GatewayLogMountPoint     = "/var/log/dynatrace/gateway"
 	GatewayTmpMountPoint     = "/var/tmp/dynatrace/gateway"
+
+	DockerImageUser  int64 = 1001
+	DockerImageGroup int64 = 1001
```
Comment (0sewa0) [inline src/controllers/dynakube/activegate/consts/consts.go:64]: just curios, why is this 1001 while the agent is 1000 ?

Comment (aorcholski) [reply src/controllers/dynakube/activegate/consts/consts.go:64]: no idea

File: src/controllers/dynakube/activegate/consts/consts.go (hunk 1 above)
Comment (aorcholski) [reply src/controllers/dynakube/activegate/consts/consts.go]: done

**Review by 0sewa0 — APPROVED** [review]

**Review by luhi-DT — APPROVED** [review]

**Pull request merged**